MantidQt/CustomInterfaces/test/ReflDataProcessorPresenterTest.h

```diff
@@ -0,0 +1,633 @@
+
+    presenter->notify(DataProcessorPresenter::ProcessFlag);
+
+    // Check output workspaces were created as expected
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsLam_13460_0_10"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsLam_13460_10_20"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsLam_13460_20_30"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsLam_13462_0_10"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsLam_13462_10_20"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsLam_13462_20_30"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsQ_13460_0_10"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsQ_13460_10_20"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsQ_13460_20_30"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsQ_13462_0_10"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsQ_13462_10_20"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsQ_13462_20_30"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist(
+        "IvsQ_13460_0_10_13462_0_10"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist(
+        "IvsQ_13460_10_20_13462_10_20"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist(
+        "IvsQ_13460_20_30_13462_20_30"));
+    TS_ASSERT(
+        AnalysisDataService::Instance().doesExist("IvsQ_binned_13460_0_10"));
+    TS_ASSERT(
+        AnalysisDataService::Instance().doesExist("IvsQ_binned_13460_10_20"));
+    TS_ASSERT(
+        AnalysisDataService::Instance().doesExist("IvsQ_binned_13460_20_30"));
+    TS_ASSERT(
+        AnalysisDataService::Instance().doesExist("IvsQ_binned_13462_0_10"));
+    TS_ASSERT(
+        AnalysisDataService::Instance().doesExist("IvsQ_binned_13462_10_20"));
+    TS_ASSERT(
+        AnalysisDataService::Instance().doesExist("IvsQ_binned_13462_20_30"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TOF_13460"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TOF_13462"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TOF_13460_0_10"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TOF_13460_10_20"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TOF_13460_20_30"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TOF_13462_0_10"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TOF_13462_10_20"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TOF_13462_20_30"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TOF_13460_monitors"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TOF_13462_monitors"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TRANS_13463"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TRANS_13464"));
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("TRANS_13463_13464"));
```

Comment:
This is easy readable. Maybe it must be like this. I would do a for loop like this:
```Python
    for (i=0, i<30; i+=10) {
      TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsQ_13460_" + to_string(i) + "_" + to_string(i+10));
      TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsQ_13462_" + to_string(i) + "_" + to_string(i+10));
      TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsLam_13460_" + to_string(i) + "_" + to_string(i+10));
      TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsLam_13462_" + to_string(i) + "_" + to_string(i+10));
      TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsQ_13460_" + to_string(i) + "_" + to_string(i+10) + "_13462_" + to_string(i) + "_" + to_string(i+10));
      TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsQ_binned_13460_" + to_string(i) + "_" + to_string(i+10));
      TS_ASSERT(AnalysisDataService::Instance().doesExist("IvsQ_binned_13462_" + to_string(i) + "_" + to_string(i+10));
      TS_ASSERT(AnalysisDataService::Instance().doesExist("TOF_13460_" + to_string(i) + "_" + to_string(i+10));
      TS_ASSERT(AnalysisDataService::Instance().doesExist("TOF_13462_" + to_string(i) + "_" + to_string(i+10));
    }
```
for parts of the assertion. It saves a lot of space. Not tested. Now the focus will be more on the actual GUI tests.

In general, I would use more `const std::string`s for some often occurring names like `IvsQ` ... `13460`.